### DISCUSSION TITLE
Moving to Another Machine

### DISCUSSION THREAD

**Discussion by eduncan911** [netbox-community/netbox-docker#476] [General]
Has anyone scripted a way to export all of the images and volumes to move and import into another machine?

I've done it a few times for one off docker apps.  There seems to be a lot going on with this docker-compose though so figured I'd ask first.

Images could be pulled down again.  However, attaching volumes requires containers to be built first - a chicken or egg thing since this compose builds both.

I'm guessing, the process would be:

- stop/shut down your working instance
- export volumes from old machine, copy to new machine
- copy your existing repository over to the new machine.
- setup a new docker compose instance on another machine, using the repo you copied (don't update to latest)

Hopefully that downloads the same image you previously had.  Unfortunately, netbox-docker isn't tagging their image.  It's always running as `latest` which could pull down a newer version.  What guarantee does `netbox-docker` give to always have an upgrade path of non-breaking changes from any previous version to `latest`?  Since I've been doing exactly that for the better part of a decade, I know how difficult that can be across dozens of services.  Hats off to the community if it's an always forward-looking auto-upgrade path (e.g. the postgres sql patches to bring it to current).  But it's extremely rare to see that in the wild.

Now I think the tricky part is that you need to detach and delete the new volumes that were just created on the new machine  Then, you're going to import the old volumes, and attach them to the existing containers you just built.

- detach and delete the old volumes
- import volumes to new machine
- attach old volumes

I haven't looked yet if anything is written to disk to be moved.  But that would be trivial - just needs to be noted.

```
$ docker images
REPOSITORY               TAG         IMAGE ID       CREATED       SIZE
redis                    6-alpine    5812dfe24a4f   12 days ago   32.3MB
postgres                 12-alpine   c2e3ad5dc14c   13 days ago   158MB
netboxcommunity/netbox   latest      1a0420c06d69   2 weeks ago   397MB
```

```
$ docker container ls
CONTAINER ID   IMAGE                           COMMAND                  CREATED          STATUS          PORTS                                             NAMES
ad1aec9c7447   netboxcommunity/netbox:latest   "/opt/netbox/docker-…"   50 seconds ago   Up 49 seconds   0.0.0.0:8000->8080/tcp, 0.0.0.0:49155->8080/tcp   netbox-docker_netbox_1
ba4153989c34   netboxcommunity/netbox:latest   "/opt/netbox/venv/bi…"   51 seconds ago   Up 49 seconds                                                     netbox-docker_netbox-worker_1
45bbcbf639b6   postgres:12-alpine              "docker-entrypoint.s…"   52 seconds ago   Up 50 seconds   5432/tcp                                          netbox-docker_postgres_1
31cfb553d501   redis:6-alpine                  "docker-entrypoint.s…"   52 seconds ago   Up 50 seconds   6379/tcp                                          netbox-docker_redis-cache_1
90544c227006   redis:6-alpine                  "docker-entrypoint.s…"   52 seconds ago   Up 50 seconds   6379/tcp                                          netbox-docker_redis_1
```

```
$ docker volume ls
DRIVER    VOLUME NAME
local     2af9a4d311c1a14be4a7bc683a87a835aa5c448c3c6026a9fec8d959a514885a
local     netbox-docker_netbox-media-files
local     netbox-docker_netbox-postgres-data
local     netbox-docker_netbox-redis-data
```

**Comment by abdros** [comment]
This is an important question. We want to move our netbox-docker from the test environment to production (two different hosts). An instruction set to do this would be extremely helpful.
